File: app/src/main/java/com/reteno/sample/fragments/ecomevents/BaseEcomEventsFragment.java
package com.reteno.sample.fragments.ecomevents;

import android.view.LayoutInflater;
import android.view.View;
import android.widget.EditText;
import android.widget.LinearLayout;

import androidx.viewbinding.ViewBinding;

import com.reteno.core.domain.model.ecom.Attributes;
import com.reteno.sample.BaseFragment;
import com.reteno.sample.R;
import com.reteno.sample.util.Util;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

class BaseEcomEventsFragment extends BaseFragment {

    protected View createNewFields(ViewBinding viewBinding, LinearLayout container) {
        return LayoutInflater
                .from(viewBinding.getRoot().getContext())
                .inflate(R.layout.view_user_custom_fields_horizontal, container, false);
    }

    protected List<Attributes> getCustomAttributes(LinearLayout llCustomAttributes) {
        int countView = llCustomAttributes.getChildCount();
        if (countView == 0) return null;

        List<Attributes> list = new ArrayList<>();
        for (int i = 0; i < countView; i++) {
            LinearLayout parent = (LinearLayout) llCustomAttributes.getChildAt(i);

            EditText etKey = (EditText) parent.getChildAt(0);
            EditText etValue = (EditText) parent.getChildAt(1);

            String key = Util.getTextOrNull(etKey);
            String value = Util.getTextOrNull(etValue);

            if (key != null && value != null) {
                List<String> valueList = Arrays.asList(value.split(","));
                list.add(new Attributes(key, valueList));
            }
        }
        return list;
    }
}

File: app/src/main/java/com/reteno/sample/fragments/ecomevents/BaseEcomEventsFragment.kt
package com.reteno.sample.fragments.ecomevents

import android.view.LayoutInflater
import android.view.View
import android.widget.EditText
import android.widget.LinearLayout
import androidx.viewbinding.ViewBinding
import com.reteno.core.domain.model.ecom.Attributes
import com.reteno.sample.BaseFragment
import com.reteno.sample.R
import com.reteno.sample.util.Util
import java.util.Arrays

open class BaseEcomEventsFragment : BaseFragment() {
    protected fun createNewFields(viewBinding: ViewBinding, container: LinearLayout?): View {
        return LayoutInflater
            .from(viewBinding.root.context)
            .inflate(R.layout.view_user_custom_fields_horizontal, container, false)
    }

    protected fun getCustomAttributes(llCustomAttributes: LinearLayout): List<Attributes>? {
        val countView = llCustomAttributes.childCount
        if (countView == 0) return null
        val list: MutableList<Attributes> = ArrayList()
        for (i in 0 until countView) {
            val parent = llCustomAttributes.getChildAt(i) as LinearLayout
            val etKey = parent.getChildAt(0) as EditText
            val etValue = parent.getChildAt(1) as EditText
            val key = Util.getTextOrNull(etKey)
            val value = Util.getTextOrNull(etValue)
            if (key != null && value != null) {
                val valueList =
                    listOf(*value.split(",".toRegex()).dropLastWhile { it.isEmpty() }
                        .toTypedArray())
                list.add(Attributes(key, valueList))
            }
        }
        return list
    }
}
